xgcm/test/test_padding.py

```diff
@@ -0,0 +1,1120 @@
+        (
+            v_face_0_padded,
+            v_face_1_padded,
+            v_face_0_padded_swapped,
+            v_face_1_padded_swapped,
+        ) = _prepad_right_left_swap_axis(v, boundary_width, fill_value, x="x", y="yl")
+
+        # Put together the additions for each face
+        u_face_0_addition = v_face_1_padded_swapped.isel(
+            yl=slice(0, boundary_width["X"][1])
+        ).rename({"x": "xl", "yl": "y"})
+        # Tangential flip (u doesnt need sign change)
+        u_face_0_addition = u_face_0_addition.isel(xl=slice(None, None, -1))
+        u_face_0_addition = _maybe_swap_dimension_names(u_face_0_addition, "y", "xl")
+
+        u_face_1_addition = v_face_0_padded_swapped.isel(
+            x=slice(
+                -boundary_width["Y"][0],
+                None if boundary_width["Y"][0] > 0 else 0,
+                # this is a bit annoying. if the boundary width on this side is
+                # 0 I want nothing to be padded. but slice(0,None) pads the whole array...
+            )
+        ).rename({"x": "xl", "yl": "y"})
+        u_face_1_addition = -u_face_1_addition.isel(y=slice(None, None, -1))
+        u_face_1_addition = _maybe_swap_dimension_names(u_face_1_addition, "y", "xl")
+
+        # now v (this one needs a sign change)
+        v_face_0_addition = u_face_1_padded_swapped.isel(
+            y=slice(0, boundary_width["X"][1])
+        ).rename({"xl": "x", "y": "yl"})
+        # Tangential flip (v DOES need sign change)
+        v_face_0_addition = -v_face_0_addition.isel(x=slice(None, None, -1))
+        v_face_0_addition = _maybe_swap_dimension_names(v_face_0_addition, "yl", "x")
+
+        v_face_1_addition = u_face_0_padded_swapped.isel(
+            xl=slice(
+                -boundary_width["Y"][0],
+                None if boundary_width["Y"][0] > 0 else 0,
+                # this is a bit annoying. if the boundary width on this side is
+                # 0 I want nothing to be padded. but slice(0,None) pads the whole array...
+            )
+        ).rename({"xl": "x", "y": "yl"})
+        v_face_1_addition = v_face_1_addition.isel(yl=slice(None, None, -1))
+        v_face_1_addition = _maybe_swap_dimension_names(v_face_1_addition, "yl", "x")
+
+        # then simply add the corresponding slice to each face according to the connection
+        u_face_0_expected = xr.concat(
+            [u_face_0_padded, u_face_0_addition],
+            dim="xl",
+        )
+        u_face_1_expected = xr.concat(
+            [u_face_1_addition, u_face_1_padded],
+            dim="y",
+        )
+
+        v_face_0_expected = xr.concat(
+            [v_face_0_padded, v_face_0_addition],
+            dim="x",
+        )
+        v_face_1_expected = xr.concat(
+            [v_face_1_addition, v_face_1_padded],
+            dim="yl",
+        )
+
+        u_expected = xr.concat([u_face_0_expected, u_face_1_expected], dim="face")
+        v_expected = xr.concat([v_face_0_expected, v_face_1_expected], dim="face")
```

Comment:
This is almost 100 lines just to generate the expected output of a single test. Perhaps we should consider just hard-coding the result arrays instead? If we keep them as small and low-dimensional as possible?

Reply:
hmm but then you have parameterized it over a lot of cases... How much of this case be factored out into one big function that generates the expected output then?